### DISCUSSION TITLE
What is the equivalent of `ethers.utils.splitSignature` in viem?

### DISCUSSION THREAD

**Discussion by bigint** [wevm/viem#458] [Question] [answered]
We stuck with the migration here https://github.com/lensterxyz/lenster/blob/main/apps/web/src/lib/splitSignature.ts#L10

**Comment by jxom** [comment]
Probably worth just using `@noble/curves` for this and derive those values from the `Signature` instance which has `r`, `s` and `v`. 
Can use `fromCompact` to instantiate, e.g:
https://github.com/wagmi-dev/viem/blob/88152e9538f35b80919c81e8be99b99291c107c3/src/utils/signature/recoverPublicKey.ts#L24

**Reply by bigint** [reply]
I tried but no luck, do you have any ideas to do it 🙇🏼 ?

**Reply by jxom** [reply, marked as answer]
```ts
import { secp256k1 } from '@noble/curves/secp256k1'

const signatureHex = '0x6e100a352ec6ad1b70802290e18aeed190704973570f3b8ed42cb9808e2ea6bf4a90a229a244495b41890987806fcbd2d5d23fc0dbe5f5256c2613c039d76db81c'
const { r, s } = secp256k1.Signature.fromCompact(signatureHex.slice(2, 130))
const v = hexToNumber(`0x${signatureHex.slice(130)}`)
```

**Reply by jxom** [reply]
Can probably make a util for this when I get some time.

**Reply by dalechyn** [reply]
@jxom this would be awesome, just stuck at it too.

**Reply by dalechyn** [reply]
For anyone wondering how to re-build signature with `r`, `s`, `v`, try that:
```
const signature = `0x${new secp256k1.Signature(
        bytesToBigint(r),
        bytesToBigint(s)
      ).toCompactHex()}${bytesToNumber(v)}
```

**Comment by grmkris** [comment]
any updates
